src/compiler/checker.ts

```diff
@@ -13679,6 +13679,9 @@ namespace ts {
                 return getLiteralType(text);
             }
             newTexts.push(text);
+            if (every(newTexts, t => t === "") && every(newTypes, t => !!(t.flags & TypeFlags.String))) {

```

Comment:
Can `newTypes` be somehow empty here? This is obviously incorrect if it can be (since the right result then is simply the empty string type). If we don't need to check for an empty type array here, we should probably at least have a comment indicating why we don't.

Reply:
There's a check for `newTypes.length === 0` literally one line above the diff cutoff here, so that definitely can't happen. Not sure a comment is necessary, when you're looking at the actual source code it's pretty clear.

Reply:
Hah, sorry, yeah, ok. Diffs don't really expand in `checker` and I was reading on mobile~